spec/test_client_script.lua

```diff
@@ -20,18 +19,34 @@ local function green_light_echo()
         print("ERROR: ", err)
         return
     end
-
+    -- Note: we should probably put this inside of the client construction/connect
     local r, err = sock:connect(HOST, PORT)
     print("INFO: Connecting to tcp socket")
     if not r then
         print("ERROR: ", err)
         return
     end
-
+    local data = "asdf"
     local config = nil
-    local websocket = ws.client(sock, "/", config):register_message_cb(
+    local recvd = false
+    local websocket = ws.client(sock, "/", config)
+    websocket:register_message_cb(
         function (msg)
-            print("INFO: Received msg: ", msg.data)
+          recvd = true
```

Comment:
this callback might need to be `fun(self: Websocket, msg: Message)` to avoid having to break the builder pattern